src/components/Newsletter/Newsletter.js

```diff
@@ -51,25 +52,35 @@ class Newsletter extends React.Component {
     });
   };
 
-  render() {
-    const { stateId, county } = this.props;
+  componentDidMount() {
+    this.handleSelectChange(this.defaultValues);
+  }
 
-    const stateDataset = US_STATE_DATASET.state_dataset.map(state => {
-      return {
-        ...state,
-        full_fips_code: state.state_fips_code + '000',
-      };
-    });
-    const countyDataset = [];
 
-    each(US_STATE_DATASET.state_county_map_dataset, (value, key) => {
-      countyDataset.push(...value.county_dataset);
+  render() {
+    const { stateId, county } = this.props;
+    this.defaultValues =  this.autocompleteOptions.filter(location => {
+      const matching_state =
+        location.state_code === stateId && location.full_fips_code.length === 2;
+      let matching_county = false;
+      if (county) {
+        matching_county =
+          location.county === county &&
+          location.state_code === stateId &&
+          location.full_fips_code.length === 5;
```

Comment:
is this length check necessary?

Reply:
mhmm otherwise this would match the state if county is null which i guess would be fine but is a lil less clear